lib/metasploit/framework/login_scanner/opnsense.rb

```diff
@@ -0,0 +1,138 @@
+require 'metasploit/framework/login_scanner/http'
+
+module Metasploit
+  module Framework
+    module LoginScanner
+
+      # This is the LoginScanner class for dealing with Deciso B.V. OPNSense instances.
+      # It is responsible for taking a single target, and a list of credentials
+      # and attempting them. It then saves the results.
+      class OPNSense < HTTP
+
+        # Retrieve the wanted cookie value by name from the HTTP response.
+        #
+        # @param [Rex::Proto::Http::Response] response The response from which to extract cookie values
+        # @param [String] wanted_cookie_name The cookie name for which to get the value
+        def get_cookie_value(response, wanted_cookie_name)
+          response.get_cookies.split('; ').find { |cookie| cookie.start_with?(wanted_cookie_name) }.split('=').last
+        end
+
+        # Checks if the target is OPNSense. The login module should call this.
+        #
+        # @return [Boolean, String] FalseClass if target is OPNSense, otherwise String
+        def check_setup
+          request_params = {
+            'method' => 'GET',
+            'uri' => normalize_uri(@uri.to_s)
+          }
+          res = send_request(request_params)
+
+          if res && res.code == 200 && res.body&.include?('Login | OPNsense')
+            return false
+          end
+
+          "Unable to locate \"Login | OPNsense\" in body. (Is this really OPNSense?)"
+        end
+
+        # Query the magic value and cookies from the OPNSense login page.
+        #
+        # @return [Hash<Symbol, Object>] A hash of the status and error or result.
+        def query_magic_value_and_cookies
+          request_params = {
+            'method' => 'GET',
+            'uri' => normalize_uri(@uri.to_s)
+          }
+
+          res = send_request(request_params)
+
+          if res.nil?
+            return { status: :failure, error: 'Did not receive response to a GET request' }
+          end
+
+          if res.code != 200
+            return { status: :failure, error: "Unexpected return code from GET request - #{res.code}" }
+          end
+
+          if res.body.nil?
+            return { status: :failure, error: 'Received an empty body from GET request' }
+          end
+
+          # The magic name and value are hidden on the login form, so we extract them using Nokogiri.
+          form_inputs = ::Nokogiri::HTML(res.body).search('input')
```

Comment:
Won't here work also this? 
```suggestion
          form_inputs = res.get_html_document&.at('input')
```